src/shared/debug/pythonDebugAdapterHeartbeat.ts

```diff
@@ -53,6 +53,7 @@ export class PythonDebugAdapterHeartbeat {
             this.socket.on('data', data => {
                 clearTimeout(timeout)
                 this.logger.verbose('Data received from Debug Adapter: %s', data.toString())
+                // we don't send a message; we assume that we always receive a telemetry event from the debugger
```

Comment:
Do we even need `waitForPythonDebugAdapter` (and `PythonDebugAdapterHeartbeat`) anymore? Or could `invokePythonLambda()` use `waitForDebugPort`, as done in `invokeCsharpLambda` and `invokeTypescriptLambda`?

```
export async function invokePythonLambda(
    ctx: ExtContext,
    config: PythonDebugConfiguration,
): Promise<PythonDebugConfiguration> {
    config.samLocalInvokeCommand = new DefaultSamLocalInvokeCommand(ctx.chanLogger, [
        WAIT_FOR_DEBUGGER_MESSAGES.PYTHON,
    ])
    config.onWillAttachDebugger = waitForDebugPort
    ...
```

Reply:
that was my initial instinct, and probably need to try it out again. my initial attempt ended in failure due to some kind of race between the port being open and actually being able to attach the debugger. Maybe that is not an issue with the `WAIT_FOR_DEBUGGER_MESSAGES` cue

Reply:
yeah, I had the same problem with ikpdb :)